### PR TITLE
Add IDA to the common interface

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
This PR implements the necessary algorithm and `solve` function to have IDA match the common interface for DAEs. The DAE solve function is similar to the ODE solve one, but there are some distinct differences in variable ordering in some of the IDA vs CVode functions which makes it too difficult to put them together (and there are some different checks that need to be done, etc.) so it's better to keep them separate. This will require the newest DiffEqBase tag.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Tests won't pass until the newest ecosystem tag goes through: https://github.com/JuliaLang/METADATA.jl/pull/7081

File: test/runtests.jl [hunk 1]
```diff
@@ -64,5 +64,5 @@ end
 println("== test common interface")
 let
 include(joinpath(examples_path, "common_interface.jl"))
-@test bool1 && bool2 && bool3 && bool4
+@test bool1 && bool2 && bool3 && bool4 && bool5 && bool6
```
Comment (musm) [inline test/runtests.jl]: I understand why you have to do this here, but later someone should think about refactoring all these tests

File: src/common.jl [hunk 2]
```diff
@@ -123,6 +124,145 @@ function solve{uType,tType,isinplace,algType<:SundialsAlgorithm,F}(
...
+        flag = @checkflag IDADense(mem, length(u0))
+
+        push!(ures, copy(u0))
+        utmp = NVector(copy(u0))
+        dutmp = NVector(copy(u0))
+        tout = [0.0]
+
+        rtest = zeros(length(u0))
+        f!(t0, u0, du0, rtest)
+        if any(abs(rtest) .>= reltol)
```
Comment (musm) [inline src/common.jl]: does this need to be `abs.` ?

Comment (ChrisRackauckas) [inline src/common.jl]: Good catch. While this would've worked in v0.5 and v0.6, might as well fix it right now. I fixed it.

**Comment by ChrisRackauckas** [comment]
The post-tag Travis passed, but started to re-run only because of `abs` -> `abs.` so I think that's safe to pull it in.

**Pull request merged**